prog/weave-npc/main.go

```diff
@@ -259,9 +259,19 @@ func root(cmd *cobra.Command, args []string) {
 	nsController := makeController(client.Core().RESTClient(), "namespaces", &coreapi.Namespace{},
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
+				defer func() {
+					if r := recover(); r != nil {
```

Comment:
could this go into a (named) subroutine?

Reply:
that would likely be cleaner, yes.  I will make this change.